### PR TITLE
Y031: Use controllerAs syntax

### PR DESCRIPTION
**Pull request by jbalboni**
This is an implementation of the Y031 style guide rule (#35). It seemed like the best path was to check that routes and states were defined with controller as syntax. This doesn't enforce anything about using 'this' inside of your controller definition instead of $scope, but that seems difficult to check.

This works for both angular-route and ui-router, though it might make sense to split them into separate rules.

This is the first eslint rule I've implemented, so if it looks like I've done something dumb, I probably have. :)

### CODE REVIEW

File: rules/utils/utils.js [hunk 1]
```diff
@@ -70,5 +70,31 @@
 		isAngularModuleDeclaration: function(node){
 			return this.isAngularComponent(node) && node.callee.type === 'MemberExpression' && node.callee.property.name === 'module'
 		},
+
+		isRouteDefinition: function (node) {
+			//the route def function is .when(), so when we find that, go up through the chain and make sure
+			//$routeProvider is the calling object
+			if (node.callee.property && node.callee.property.name === 'when') {
+				var callObject = node.callee.object;
+				while (!callObject.type || callObject.type !== 'Identifier') {
```
Comment (jbalboni) [inline rules/utils/utils.js]: Could do this recursively instead.

Comment (Gillespie59) [inline rules/utils/utils.js]: You mean in order to get this use case ? 

```
$routeProvider.when(...).when(...)
```

Comment (jbalboni) [inline rules/utils/utils.js]: Yeah. It's already handling that use case, but you could do it recursively instead of with a while loop.

Comment (Gillespie59) [inline rules/utils/utils.js]: Yes we can make it recurive. Would you update your PR ? thanks ;)

**Comment by jbalboni** [comment]
Yes, that makes sense to me. Do you think that $scope check should be implemented in this same eslint rule, or in a separate one (the docs could indicate they were both part of Y031 if they were separate)?

**Comment by Gillespie59** [comment]
I will maybe rename your rule into ng_router_controller_as. And the '$scope' check will be named ng_controller_as.
is it OK for you ?

**Comment by jbalboni** [comment]
Yep, makes sense. I can change the name when I make the recursion change, if you want.

**Comment by jbalboni** [comment]
I've made the updates discussed, including the name change.

**Comment by Gillespie59** [comment]
Thanks a lot !!

**Pull request merged**